src/plugins/intel_cpu/src/emitters/x64/jit_perf_count_emitters.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "jit_emitter.hpp"
+#include "jit_perf_count_emitters.hpp"
+#include <cpu/x64/jit_generator.hpp>
+
+using namespace dnnl::impl;
+using namespace dnnl::impl::utils;
+using namespace dnnl::impl::cpu;
+using namespace dnnl::impl::cpu::x64;
+using namespace Xbyak;
+using namespace Xbyak::util;
+
+namespace ov {
+namespace intel_cpu {
+
+#ifdef CHRONO_CALL
+static void get_current_time(std::chrono::high_resolution_clock::time_point* current_time) {
+    *current_time = std::chrono::high_resolution_clock::now();
+}
+
+static void get_accumulated_time(std::chrono::high_resolution_clock::time_point* start_time, uint64_t* accumulation, uint32_t* num) {
+    auto current_time = std::chrono::high_resolution_clock::now();
+    *accumulation += std::chrono::duration_cast<std::chrono::nanoseconds>(current_time - *start_time).count();
+    (*num)++;
+}
+#endif
+
+jit_perf_count_start_emitter::jit_perf_count_start_emitter(dnnl::impl::cpu::x64::jit_generator *host, dnnl::impl::cpu::x64::cpu_isa_t host_isa,
+                                            const std::shared_ptr<ov::Node>& n) : jit_emitter(host, host_isa) {
+    auto start_op = ov::as_type_ptr<snippets::op::PerfCountBegin>(n);
+#ifdef CHRONO_CALL
+    m_current_time = &(start_op->start_time_stamp);
```

Comment:
@IvanNovoselov I used to try this manner, but failed as `PerfCountBegin->start_time()` or `PerfCountEnd->stop_time()`  should be non-static function, which need change private time variable. h->call for class static member function is ok, but failed for non-static function with pointer of member cast problem. Now I create a wrapper to cast successfully, will validate it if it work. Please correct me if call non-static function is not correct from theory analysis perspective.

Reply:
@IvanNovoselov, As rdtsc based emitters is for lightest overhead. Internal call(for example get/set functions) should be totally avoided. so access public member is needed. what do you think?

Reply:
> @IvanNovoselov I used to try this manner, but failed as `PerfCountBegin->start_time()` or `PerfCountEnd->stop_time()` should be non-static function, which need change private time variable. h->call for class static member function is ok, but failed for non-static function with pointer of member cast problem. Now I create a [wrapper to cast](PerfCountBegin-%3Estop_time()) successfully, will validate it if it work. Please correct me if call non-static function is not correct from theory analysis perspective.

I'm not sure I fully understand the issue here. If you store a pointer to node in an emitter (as done in LoopEndEmitter for example), then you can call any non-static functions for this node from emit_impl since the pointer itself is not changed. For example, you can `loop_end->set_increment()` from `LoopEndEmitter::emit_impl`. Although this is not a good practice and should generally be avoided (because the graph can be used after generation), however this approach is tolerable for selected operations is we update only internal variable (and don't change ops connectivity etc).

Reply:
> @IvanNovoselov, As rdtsc based emitters is for lightest overhead. Internal call(for example get/set functions) should be totally avoided. so access public member is needed. what do you think?
Right, as we discussed, the general idea here is to make two perf counter operations: one based on std::shrono the second one is based on rdtsc. The chroho impl provides universal solution, it is backend-independent and should be placed in snippets opset, and we should try to simplify interface of this impl even at a cost of couple extra dereferencings.

The rtdsc impl is another thing: it uses x86-specific features and thus allows for smaller overheads and more accurate measurements. It also should be placed in the plugin part of snippets opset. So getting back to your question, yes, I think it should be Ok to access public fields in this case (as done in LoopBegin/LoopEnd for example).

Reply:
> > @IvanNovoselov I used to try this manner, but failed as `PerfCountBegin->start_time()` or `PerfCountEnd->stop_time()` should be non-static function, which need change private time variable. h->call for class static member function is ok, but failed for non-static function with pointer of member cast problem. Now I create a [wrapper to cast](PerfCountBegin-%3Estop_time()) successfully, will validate it if it work. Please correct me if call non-static function is not correct from theory analysis perspective.
> 
> I'm not sure I fully understand the issue here. If you store a pointer to node in an emitter (as done in LoopEndEmitter for example), then you can call any non-static functions for this node from emit_impl since the pointer itself is not changed. For example, you can `loop_end->set_increment()` from `LoopEndEmitter::emit_impl`. Although this is not a good practice and should generally be avoided (because the graph can be used after generation), this approach is still tolerable for selected operations since we update only internal variable (and don't change the op connectivity etc).

you are right, call non-static member function of node from pointer of node successfully. thanks!

Reply:
> > @IvanNovoselov, As rdtsc based emitters is for lightest overhead. Internal call(for example get/set functions) should be totally avoided. so access public member is needed. what do you think?
> 
> Right, as we discussed, the general idea here is to make two perf counter operations: one based on std::shrono the second one is based on rdtsc. The chroho impl provides universal solution, it is backend-independent and should be placed in snippets opset, and we should try to simplify interface of this impl even at a cost of couple extra dereferencings.
> 
> The rtdsc impl is another thing: it uses x86-specific features and thus allows for smaller overheads and more accurate measurements. It also should be placed in the plugin part of snippets opset. So getting back to your question, yes, I think it should be Ok to access public fields in this case (as done in LoopBegin/LoopEnd for example).

Created separately. rdtsc is locate in cpu plugin here [src/plugins/intel_cpu/src/transformations/snippets/x64/op/perf_count_rdtsc.hpp](https://github.com/openvinotoolkit/openvino/pull/19493/files#diff-e53ad3449538575135fc799c9aea13bfe29d79f7b812acaf3bc0ab7e327d57fc), could you please chech is this the right place? thanks